fatiando/seismic/conv.py

```diff
@@ -0,0 +1,251 @@
+    >>> synt = conv.convolutional_model(rc, 30., conv.rickerwave)
+    >>> # Plot the result
+    >>> fig = plt.figure(figsize=(6,5))
+    >>> _ = mpl.seismic_wiggle(synt, dt=2.e-3)
+    >>> _ = mpl.seismic_image(synt, dt=2.e-3,
+    ...                            cmap=mpl.pyplot.cm.jet, aspect='auto')
+    >>> _ = plt.ylabel('time (seconds)')
+    >>> _ = plt.title("Convolutional seismogram", fontsize=13, family='sans-serif')
+
+References
+----------
+
+Yilmaz, Oz,
+Ch.2 Deconvolution. In: YILMAZ, Oz. Seismic Data Analysis: Processing,
+Inversion, and Interpretation of Seismic Data. Tulsa: Seg, 2001. Cap. 2.
+p. 159-270. Available at: <http://dx.doi.org/10.1190/1.9781560801580.ch2>
+
+
+"""
+from __future__ import division
+import numpy as np
+from scipy import interpolate  # linear interpolation of velocity/density
+
+
+def convolutional_model(rc, f, wavelet, dt=2.e-3):
+    """
+    Calculate convolutional seismogram for a geological model
+
+    Calculate the synthetic convolutional seismogram of a geological model, Vp
+    is mandatory while density is optional. The given model in a matrix form is
+    considered a mesh of square cells.
+
+    .. warning::
+
+        Since the relative difference between the model is the important, being
+        consistent with the units chosen for the parameters is the only
+        requirement, whatever the units.
+
+    Parameters:
+
+    * rc : 2D-array
+        reflectivity values in the time domain
+    * f : float
+        Dominant frequency of the ricker wavelet
+    * wavelet : float
+        The function to consider as source in the seismic modelling.
+    * dt: float
+        Sample time of the ricker wavelet and of the resulting seismogram
+
+    Returns:
+
+    * synth_l : 2D-array
+        Resulting seismogram
+
+    """
+    w = wavelet(f, dt)
+    # convolution
+    synth_l = np.zeros(np.shape(rc))
+    for j in range(0, rc.shape[1]):
+        if np.shape(rc)[0] >= len(w):
+            synth_l[:, j] = np.convolve(rc[:, j], w, mode='same')
+        else:
+            aux = np.floor(len(w)/2.)
+            synth_l[:, j] = np.convolve(rc[:, j], w, mode='full')[aux:-aux]
+    return synth_l
+
+
+def reflectivity(model_t, rho=1.):
+    """
+    Calculate reflectivity series
+
+    Parameters:
+
+    * model_t : 2D-array
+        Vp values in time domain
+    * rho : 2D-array (optional)
+        Density values for all the model, in time domain.
+
+    Returns:
+
+    * rc : 2D-array
+        Calculated reflectivity series for all the model given.
+    """
+    rc = np.zeros(np.shape(model_t))
+    # dimension of rho must be the same of velocity grid, if both are matrix
+    try:
+        rc[1:, :] = ((model_t[1:, :]*rho[1:, :]-model_t[:-1, :]*rho[:-1, :]) /
+                     (model_t[1:, :]*rho[1:, :]+model_t[:-1, :]*rho[:-1, :]))
+    except TypeError:
+        rc[1:, :] = ((model_t[1:, :]-model_t[:-1, :]) /
+                     (model_t[1:, :]+model_t[:-1, :]))
```

Comment:
@leouieda Rewriting the code I noticed one thing. Maybe the user is interested in calculate the reflectivity just for one trace. 
So, how do you think is the best way to keep this option (if you think it a good idea)? I realized to keep the `try`block, but in the `except` I would write considering `model_t` and `rho` as one dimensional. In this way,  `assert model_t.shape == rho.shape`  would replace `assert rho.ndim == 2`.